WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayInformationViewModel.swift

```diff
@@ -55,6 +61,20 @@ final class SetUpTapToPayInformationViewModel: PaymentSettingsFlowPresentedViewM
         stores.dispatch(connectedAction)
     }
 
+    private func beginConnectivityObservation() {
+        connectivityObserver.statusPublisher.sink { [weak self] status in
+            guard let self = self else { return }
+            switch (status, self.enableSetup) {
+            case (.notReachable, true),
+                (.reachable, false):
+                self.enableSetup.toggle()
```

Comment:
TIL we can build a switch like this.

Reply:
@iamgabrielma Yep, tuples in switches can be super helpful! You can also ignore one or other side of the tuple in a pattern match, e.g.

```suggestion
            case (.notReachable, true),
                (.reachable, false),
                (.unknown, _):
                self.enableSetup.toggle()
```

Which would toggle the `enableSetup` whether the current state was true or false if the reachability state is unknown. Not something we would actually do here!